lib/Driver/ToolChains.h

```diff
@@ -40,6 +40,12 @@ class LLVM_LIBRARY_VISIBILITY Darwin : public ToolChain {
 
   bool shouldStoreInvocationInDebugInfo() const override;
 
+private:
```

Comment:
If these are never called outside of DarwinToolChains.cpp, should they just be static functions instead, perhaps with an extra argument for the `Darwin` instance? (This is genuinely a question; I'm not sure if that would be better.)

Reply:
C++ makes it tricky to do that if the helper needs to access any private members, but if it doesn't then I agree.

Reply:
Heh. Brent meant the other kind of `static`, i.e. a local function in the .cpp file.

Reply:
Ohh I see, not so hard to change.